concept/type_/relation_type.rs

```diff
@@ -257,21 +290,33 @@ impl<'a> RelationType<'a> {
         &self,
         snapshot: &impl ReadableSnapshot,
         type_manager: &'m TypeManager,
-    ) -> Result<MaybeOwns<'m, HashMap<RoleType<'static>, Relates<'static>>>, ConceptReadError> {
+    ) -> Result<MaybeOwns<'m, HashSet<Relates<'static>>>, ConceptReadError> {
         type_manager.get_relation_type_relates(snapshot, self.clone().into_owned())
     }
 
-    // TODO: It looks like a hack to me right now..... Why don't we search it, but build a new one?
-    pub fn get_relates_of_role(
+    pub fn get_relates_role(
+        &self,
+        snapshot: &impl ReadableSnapshot,
+        type_manager: &TypeManager,
+        role_type: RoleType<'static>,
+    ) -> Result<Option<Relates<'static>>, ConceptReadError> {
+        Ok(self.get_relates(snapshot, type_manager)?.iter().find(|relates| relates.role() == role_type).cloned())
+    }
+
+    pub fn get_relates_role_name(
         &self,
         snapshot: &impl ReadableSnapshot,
         type_manager: &TypeManager,
         role_name: &str,
     ) -> Result<Option<Relates<'static>>, ConceptReadError> {
-        let role_label = Label::build_scoped(role_name, self.get_label(snapshot, type_manager)?.name().as_str());
-        Ok(type_manager
-            .get_role_type(snapshot, &role_label)?
-            .map(|role_type| Relates::new(self.clone().into_owned(), role_type)))
+        for relates in type_manager.get_relation_type_relates(snapshot, self.clone().into_owned())?.into_iter() {
```

Comment:
Just double check, I think Krishnan or Dmitrii recently added an API to get relations with a role Name, which might not be useful here (about as expensive/difficult) but might be useful elsewhere

Reply:
Yes, I see `get_roles_by_name`, but it's different and is not scoped to specific relation. Thanks